cpp/include/cudf/functions.h

```diff
@@ -927,6 +928,65 @@ gdf_error gdf_quantile_aprrox(	gdf_column*  col_in,       //input column with 0
                                 void*        t_erased_res, //type-erased result of same type as column;
                                 gdf_context* ctxt);        //context info
 
+
+/* --------------------------------------------------------------------------*/
+  /**
+   * @brief Sorts a set of columns based on specified "key" columns. Returns a column containing
+   * the offset to the start of each group
+   *
+   * @param[in]  num_data_cols         The number of columns in the dataset
+   * @param[in]  data_cols_in          The input columns in the dataset
+   * @param[in]  num_key_cols      The number of key columns
+   * @param[in]  key_col_indices   The column indices of the input dataset that are they keys by which the data will be grouped by
+   * @param[out] data_cols_out         The dataset sorted by the group by columns (needs to be pre-allocated. Should be the same number of columns and rows as data_cols_in)
+   * @param[out] group_start_indices   An array containing the starting indices of each key (group). (needs to be pre-allocated to have the same number of rows as data_cols_in)
+   * @param[out] num_group_start_indices   The number of elements in group_start_indices
+   * @param[in]  ctxt                  The context used to control how nulls are treated in a sort and in group by
+   *   ctxt->flag_nulls_sort_behavior
+   *    GDF_NULL_AS_LARGEST = Nulls are are treated as largest, 
+   *    GDF_NULL_AS_SMALLEST = Nulls are treated as smallest, 
+   *    GDF_NULL_AS_LARGEST_FOR_MULTISORT = Special multicolumn-sort case. A row with null in any column is largest>
+   *   ctxt-> flag_groupby_include_nulls 
+   *    false = Nulls are ignored in group by keys (Pandas style),
+   *    true = Nulls are treated as values in group by keys where NULL == NULL (SQL style)>
+   *
+   * @returns gdf_error with error code on failure, otherwise GDF_SUCESS
+   */
+  /* ----------------------------------------------------------------------------*/
+gdf_error gdf_group_by_without_aggregations(gdf_size_type num_data_cols,
```

Comment:
Mark, I talked about this with Jake. He had the same sort of comment. I think that its all a matter of perspective. In the SQL world when you have a group by, you always have a reduction, but this function does not reduce the data set, it just sorts it and returns some indices of where groups start, so it does not really feel like a group by to me, which is why I personally like the name gdf_group_by_without_aggregations because its very explicit. My other concern is if it will confuse and/or clash with existing function names. For example, there is already a function called gdf_group_by in the file new_groupby.hpp which I believe currently handles all the hash based group by functions gdf_group_by_sum, gdf_group_by_min, etc.. Now I care more about being able to close out this PR and not spend more cycles on it. If you feel strongly enough about the name, we can change it, but I would request we change it once all the other group_by related work is done (Jake's rework, and the sort based group by implementation) and we can rework the names all at once.